src/sudoers/mod.rs

```diff
@@ -189,7 +191,37 @@ fn group_cmd_specs_per_runas<'a>(
             runas = Some(new_runas);
         }
 
-        collected_specs.push((tag, spec));
+        let (negate, meta) = match spec {
+            Qualified::Allow(meta) => (false, meta),
+            Qualified::Forbid(meta) => (true, meta),
+        };
+
+        if let Meta::Alias(alias_name) = meta {
+            if let Some(specs) = cmnd_aliases.get(alias_name) {
+                // expand Cmnd_Alias
+                for spec in specs.iter() {
+                    let (allow, meta) = match spec {
+                        Qualified::Allow(meta) => (true, meta),
+                        Qualified::Forbid(meta) => (false, meta),
+                    };
+
+                    let spec = if allow ^ negate {
+                        Qualified::Allow(meta)
+                    } else {
+                        Qualified::Forbid(meta)
+                    };
+
+                    collected_specs.push((tag.clone(), spec))
+                }
+            }
+        } else {
+            let spec = match spec {
```

Comment:
Is this because you need to do some magic conversion due to generics or something?

Reply:
`list` is an iterator of items of type `&Qualified<Meta<_>>`. because we may need to negate the `Qualified` enum in the `if` branch, `collected_specs` must have type `Vec<Qualified<&Meta<_>>>` (note the position of the reference changed). in this `else` branch we must do the same type conversion from `&Qualified<Meta<_>>` to `Qualified<&Meta<_>>` -- this is effectively like `Option::as_ref` so maybe I should add a `Qualified::as_ref` method.